/**
 * Copyright (c) 2014-present, The osquery authors
 *
 * This source code is licensed as defined by the LICENSE file found in the
 * root directory of this source tree.
 *
 * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
 */

syntax = "proto3";

package containerd.events;

import "google/protobuf/any.proto";

message ContainerCreate {
	string id = 1;
	string image = 2;
	message Runtime {
		string name = 1;
		google.protobuf.Any options = 2;
	}
	Runtime runtime = 3;
}

message ContainerUpdate {
	string id = 1;
	string image = 2;
	map<string, string> labels  = 3;
	string snapshot_key = 4;
}

message ContainerDelete {
	string id = 1;
}